### ISSUE TITLE
About the Constraints of export annotation task

### ISSUE THREAD

**Issue by Pan-will** [opencv/cvat#6298]
### Briefly summarize the problem

Excuse me. I wanted to export the annotation task, but every time I created a new project, I found that it didn't export xml immediately, and by the next morning it was! Does CVAT set any restrictions on the export tasks? For example, it can't be exported until 12 hours have passed, or it can't be exported until all jobs in the task have been reviewed.
If there is a restriction, should you remove it on the frontend or backend?

### Detailed operation steps

1. New annotation project：The Cat Annotation.
2. Create a new annotation task under this project: Cat Annotation Task.
3. The source files for this task were pulled from the cloud storage (S3 used) and contained 100 images of cats. Divided into 10 jobs, 10 images/jobs.
4. Complete each annotation job.
5. Go to the Task list page, select the ‘export to dataset’ function of the ‘Cat Annotation Task’ task, export the location to the cloud storage (using S3), and submit the form.
6. The odd thing happens: A dialog tells me : the export will start automatically once all the annotation jobs are ready.
In fact, I've already done all the annotation jobs for this task.

## To summarize my question

1. Are there any limitations to the annotation task export function of the CVAT system?
2. If there are restrictions, where should I remove them in the frontend file, or in a class on the backend?

If you can help with the answer, I would really appreciate it!

**Comment by Pan-will** [comment]
Export annotation task calls the backend API is:  /api/tasks/93/annotations
A set of parameters I've used:
use_default_location=false&location=cloud_storage&cloud_storage_id=1&filename=1686217798055pqklvy.zip&format=CVAT+for+images+1.1

**Comment by bsekachev** [comment]
@Pan-will 

> it didn't export xml immediately

Yes, it did not because exporting in general is a long operation for huge projects. May take even hours. It is why there are **queues** on the server.

> by the next morning it was

Probably the queues are busy when you are trying to export your task. In morning they are usually less loaded if we are talking about app.cvat.ai. You can just click "export", keep the page opened and go for a walk, tea, or sleep. Do not need to wait in front of PC. If you exporing to S3, you can switch off the PC immediately. 

> For example, it can't be exported until 12 hours have passed, or it can't be exported until all jobs in the task have been reviewed.

There is not any restrictions. Only performance issues described above. 

> The odd thing happen

The message is: "Annotations export was started for task #25. Download will start automatically as soon as the Annotations is ready." 

Probably we could improve it and make it different for S3 storage and local downloading. Basically it is the same for both targets. The message actually means that server needs some time to prepare the dataset and after this browser will start downloading the archive.

**Comment by Pan-will** [comment]
@bsekachev 
Thank you very much for your patience!
In the CVAT system, is the queue you mentioned queue using redis transactions?

By the way: I'm working on an interesting "automated model training" project that includes milestones on the link such as data annotation, dataset generation, model library, model training, service publishing, and more! CVAT is used for data annotation! This is interesting work, if successful, I will abstract the workflow and share the relevant parts of the annotation here!

**Comment by bsekachev** [comment]
Yes, queues use redis-server. 
``cvat_worker_export`` service is responsible for exporting. 

>  I will abstract the workflow and share the relevant parts of the data annotation here!

Sounds interesting!

**Comment by bsekachev** [comment]
BTW, on self-hosted solution you can look at queues in admin panel here: ``<host>/django-rq/``
<img width="1642" alt="image" src="https://github.com/opencv/cvat/assets/40690378/3e1f690d-1e0a-47b8-bde2-04f80d0614ff">
